src/django_safemigrate/__init__.py

```diff
@@ -1,7 +1,33 @@
+from dataclasses import dataclass
+from datetime import timedelta
 from enum import Enum
+from typing import Optional
 
 
-class Safe(Enum):
+class SafeEnum(Enum):
     always = "always"
     before_deploy = "before_deploy"
     after_deploy = "after_deploy"
+
+
+@dataclass
+class Safe:
+    safe: SafeEnum
+    delay: Optional[timedelta] = None
+
+    @classmethod
+    def always(cls):
+        return cls(safe=SafeEnum.always)
+
+    @classmethod
+    def before_deploy(cls):
+        return cls(safe=SafeEnum.before_deploy)
+
+    @classmethod
+    def after_deploy(cls, *, delay: timedelta = None):
+        return cls(safe=SafeEnum.after_deploy, delay=delay)
+
+    def __eq__(self, other):
+        if isinstance(other, self.__class__):
+            return other.safe == self.safe and other.delay == self.delay
+        raise TypeError()
```

Comment:
Is there something inappropriate about the automatically generated `__eq__` method?

Reply:
Nope, I can remove it since we're not using it.

Reply:
Sounds good. My impressions was that the autogenerated one should work pretty well even if we were.